src/client/Microsoft.Identity.Client/PlatformsCommon/Shared/SimpleHttpClientFactory.cs

```diff
@@ -14,23 +16,66 @@ namespace Microsoft.Identity.Client.PlatformsCommon.Shared
     /// .NET should use the IHttpClientFactory, but MSAL cannot take a dependency on it.
     /// .NET should use SocketHandler, but UseDefaultCredentials doesn't work with it 
     /// </remarks>
-    internal class SimpleHttpClientFactory : IMsalHttpClientFactory
+    internal class SimpleHttpClientFactory : IMsalMtlsHttpClientFactory
     {
         //Please see (https://aka.ms/msal-httpclient-info) for important information regarding the HttpClient.
-        private static readonly Lazy<HttpClient> s_httpClient = new Lazy<HttpClient>(InitializeClient);
+        private static readonly ConcurrentDictionary<string, HttpClient> s_httpClient = new ConcurrentDictionary<string, HttpClient>();
+        private static readonly object s_cacheLock = new object();
 
-        private static HttpClient InitializeClient()
+        private static HttpClient CreateNonMtlsClient()
         {
-            var httpClient = new HttpClient(new HttpClientHandler() { 
-                /* important for IWA */ UseDefaultCredentials = true });
+            var httpClient = new HttpClient(new HttpClientHandler()
+            {
+                /* important for IWA */
+                UseDefaultCredentials = true
+            });
+            HttpClientConfig.ConfigureRequestHeadersAndSize(httpClient);
+
+            return httpClient;
+        }
+
+        private static HttpClient CreateMtlsHttpClient(X509Certificate2 bindingCertificate)
+        {
+            if (s_httpClient.Count > 1000)
+                CheckAndClearCache();
+
+            //Create an HttpClientHandler and configure it to use the client certificate
+            HttpClientHandler handler = new();
+            
+#if SUPPORTS_MTLS
+            handler.ClientCertificates.Add(bindingCertificate);
+#endif
+            var httpClient = new HttpClient(handler);
             HttpClientConfig.ConfigureRequestHeadersAndSize(httpClient);
 
             return httpClient;
         }
 
         public HttpClient GetHttpClient()
         {
-            return s_httpClient.Value;
+            return s_httpClient.GetOrAdd("non_mtls", CreateNonMtlsClient());
+        }
+
+        public HttpClient GetHttpClient(X509Certificate2 x509Certificate2)
+        {
+            if (x509Certificate2 == null)
+            {
+                return GetHttpClient();
+            }
+
+            string key = x509Certificate2.Thumbprint;
+            return s_httpClient.GetOrAdd(key, CreateMtlsHttpClient(x509Certificate2));
+        }
+
+        private static void CheckAndClearCache()
+        {
+            lock (s_cacheLock)
+            {
+                if (s_httpClient.Count > 1000)
+                {
+                    s_httpClient.Clear();
```

Comment:
I don't know about this. If there are pending requests on other threads they will be cancelled. The better solution would be to use an LRU cache, but that's probably a bit overkill.  

Can you log the size?